history-service/test/test.js

```diff
@@ -0,0 +1,51 @@
+import chai, { assert } from "chai"
+import chaiHttp from "chai-http"
+import app from "../index.js"
+import { userHistory } from "./testData.js"
+
+chai.use(chaiHttp)
+chai.should()
+
+describe("history-service tests", () => {
+    describe("POST/create-history", () => {
+        it("Create history successful", (done) => {
+            chai.request(app)
+                .post("/create-history")
+                .send(userHistory)
+                .end((err, res) => {
+                    res.should.have.status(200)
+                    res.body.should.be.a("object")
+                    res.body.should.have.property("message")
+                    assert.equal(res.body.message, `interview history saved for ${userHistory.username}`)
+                    done()
+                })
+        })
+    })
+
+    describe("GET/get-history/:username", () => {
+        it("Get history succeessful", (done) => {
+            chai.request(app)
+            .get(`/get-history/${userHistory.username}`)
+            .end((err, res) => {
+                res.should.have.status(200)
+                res.body.should.be.a("array")
+                res.body[0].should.have.property("username")
+                assert.equal(res.body[0].username, "test1")
+                res.body[0].should.have.property("matchUsername")
+                assert.equal(res.body[0].matchUsername, "test2")
+                res.body[0].should.have.property("difficulty")
+                assert.equal(res.body[0].difficulty, "easy")
+                res.body[0].should.have.property("question")
+                res.body[0].should.have.nested.property("question.difficulty")
+                assert.equal(res.body[0].question.difficulty, "easy")
+                res.body[0].should.have.nested.property("question.title")
+                assert.equal(res.body[0].question.title, "test")
+                res.body[0].should.have.nested.property("question.description")
+                assert.equal(res.body[0].question.description, "test")
+                res.body[0].should.have.nested.property("question.link")
+                assert.equal(res.body[0].question.link, "test")
+                done()
+            })
+        })
+    })
+})
```

Comment:
Please add after hook to remove occurrences where username is test1. This is because when you run `npm test` multiple times it will keep creating entries.

For example:

```
after(async () => {
            await History.deleteMany({username: "test1"})
        })
```

I used after hooks here https://github.com/CS3219-AY2223S1/cs3219-project-ay2223s1-g59/blob/main/matching-service/test/test.js

Reply:
Ok but technically if the user creates a user called tes1 then it will get deleted.

Reply:
Updated to remove test data by taking the _id created for the history object and removing that particular history object